### ISSUE TITLE
jacobian_warping computation of right-multiplication

### ISSUE THREAD

**Issue by 447425299** [lvzhaoyang/DeeperInverseCompositionalAlgorithm#4]
Thanks for your great work. I have a question related to the jacobian_warping：
In models/algorithms.py, in the function
def compute_jacobian_warping(p_invdepth, K, px, py):
...
    dx_dp = torch.cat((-xy,     1+x**2, -y, invd, O, -invd*x), dim=2)
    dy_dp = torch.cat((-1-y**2, xy,     x, O, invd, -invd*y), dim=2)
...
The jacobian (dx_dp,dy_dp) here is the left-multiplication jacobian.
Are the left-multiplication and right-multiplication jacobian the same?
I am not familiar with right-multiplication, Could you explain it? Thanks a lot.

**Comment by lvzhaoyang** [comment]
No, it won't be the same. The jacobian of the right-multiplication will requires an additional adjoint operation of the Lie-group, which you can refer to http://www.ethaneade.org/lie.pdf, section 2.3, 3.3, & 3.4.

**Comment by 447425299** [comment]
Thank you very much for sharing and helping me to deepen my understanding of Lie-Group.
However, according to my derivation, the {Delta xi}^{-1} calculated according to formulas (28) and (11) in the paper should be multiplied to xi_k by left-multiplication. I usually use left-multiplication.  the Jacobian is the same as (28). Is there any problem with me?  Thank you very much for your answer.

**Comment by zhaoyang-lv** [comment]
Not quite sure what is your derivation. But the original equation (5) does not have xi_k to be part of the chain rule. I think you may solve the jacobian from a different linearization.

**Comment by 447425299** [comment]
Thank you zhaoyang, I've figured it out. I confused the optical flow method with the direct method before.
